File: music-lessons-2-java/app/src/main/java/com/mstoyanov/musiclessons/model/Student.java
package com.mstoyanov.musiclessons.model;

import android.arch.persistence.room.ColumnInfo;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.Ignore;
import android.arch.persistence.room.PrimaryKey;
import android.support.annotation.NonNull;

import java.io.Serializable;
import java.util.List;

@Entity(tableName = "student")
public class Student implements Comparable<Student>, Serializable {
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "_id")
    private long studentId;
    @ColumnInfo(name = "first_name")
    private String firstName;
    @ColumnInfo(name = "last_name")
    private String lastName;
    private String email;
    private String notes;
    @Ignore
    private List<PhoneNumber> phoneNumbers;

    public long getStudentId() {
        return studentId;
    }

    public void setStudentId(long studentId) {
        this.studentId = studentId;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getNotes() {
        return notes;
    }

    public void setNotes(String notes) {
        this.notes = notes;
    }

    public List<PhoneNumber> getPhoneNumbers() {
        return phoneNumbers;
    }

    public void setPhoneNumbers(List<PhoneNumber> phoneNumbers) {
        this.phoneNumbers = phoneNumbers;
    }

    @Override
    public int compareTo(@NonNull Student s) {
        int result = firstName.compareToIgnoreCase(s.firstName) != 0 ?
                firstName.compareToIgnoreCase(s.firstName) :
                lastName.compareToIgnoreCase(s.lastName);
        return result;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Student student = (Student) o;

        if (studentId != student.studentId) return false;
        if (firstName != null ? !firstName.equals(student.firstName) : student.firstName != null)
            return false;
        if (lastName != null ? !lastName.equals(student.lastName) : student.lastName != null)
            return false;
        if (email != null ? !email.equals(student.email) : student.email != null) return false;
        return notes != null ? notes.equals(student.notes) : student.notes == null;
    }

    @Override
    public int hashCode() {
        int result = (int) (studentId ^ (studentId >>> 32));
        result = 31 * result + (firstName != null ? firstName.hashCode() : 0);
        result = 31 * result + (lastName != null ? lastName.hashCode() : 0);
        result = 31 * result + (email != null ? email.hashCode() : 0);
        result = 31 * result + (notes != null ? notes.hashCode() : 0);
        return result;
    }

    @Override
    public String toString() {
        return "Student{" +
                "studentId=" + studentId +
                ", firstName='" + firstName + '\'' +
                ", lastName='" + lastName + '\'' +
                ", email='" + email + '\'' +
                ", notes='" + notes + '\'' +
                ", phoneNumbers=" + phoneNumbers +
                '}';
    }
}

File: music-lessons-2-kotlin/app/src/main/java/com/mstoyanov/musiclessons/model/Student.kt
package com.mstoyanov.musiclessons.model

import android.arch.persistence.room.ColumnInfo
import android.arch.persistence.room.Entity
import android.arch.persistence.room.Ignore
import android.arch.persistence.room.PrimaryKey
import java.io.Serializable

@Entity(tableName = "student")
data class Student @Ignore constructor(@PrimaryKey(autoGenerate = true) @ColumnInfo(name = "_id") var studentId: Long = 0,
                                       @ColumnInfo(name = "first_name") var firstName: String? = null,
                                       @ColumnInfo(name = "last_name") var lastName: String? = null,
                                       var email: String? = null,
                                       var notes: String? = null,
                                       @Ignore var phoneNumbers: MutableList<PhoneNumber>? = null) : Comparable<Student>, Serializable {

    constructor() : this(0, null, null, null, null, null)

    override fun compareTo(other: Student): Int {
        return if (firstName!!.compareTo(other.firstName!!, ignoreCase = true) != 0)
            firstName!!.compareTo(other.firstName!!, ignoreCase = true)
        else
            lastName!!.compareTo(other.lastName!!, ignoreCase = true)
    }
}